Content.Server/DeviceLinking/Systems/SignalSwitchSystem.cs

```diff
@@ -38,8 +38,18 @@ private void OnActivated(EntityUid uid, SignalSwitchComponent comp, ActivateInWo
             _deviceLink.SendSignal(uid, comp.StatusPort, comp.State);
         }
 
+        RaiseLocalEvent(uid, new SwitchPressedEvent());
+
         _audio.PlayPvs(comp.ClickSound, uid, AudioParams.Default.WithVariation(0.125f).WithVolume(8f));
 
         args.Handled = true;
     }
 }
+
+// Event to inform any other systems that may want to know that the signal switch was pressed
+public sealed class SwitchPressedEvent : EntityEventArgs
```

Comment:
New events should be record structs raised by ref if possible.

Reply:
yep
```cs
[ByRefEvent]
public record struct SwitchPressedEvent();
```